### ISSUE TITLE
Read entire NULL object column

### ISSUE THREAD

**Issue by Natalie-Caruana** [awslabs/aws-data-wrangler#524]
Hi, When I try to read the following dataframe from an s3 bucket:
```
dataframe = pd.DataFrame({
    "col": [1, 2],
    "col_string_null": [None, None],
    "col_date_null": [None, None],
})
```
as an Iteratable and as dataframe
```
import awswrangler as wr
df = wr.s3.read_parquet(path='s3://<S3_BUCKET>/df_null_columns.parquet', boto3_session=boto3.Session())
gen = wr.s3.read_parquet(path='s3://<S3_BUCKET>/df_null_columns.parquet', chunked=True,boto3_session=boto3.Session())
next(gen)
```
I get the following error for the iterable

```
packages/awswrangler/_data_types.py", line 188, in pyarrow2athena
    raise exceptions.UndetectedType("We can not infer the data type from an entire null object column")
awswrangler.exceptions.UndetectedType: We can not infer the data type from an entire null object column
```

Is there a way in which i can get the iterable from a dataframe containing entire NULL object columns?

Thanks!

**Comment by maxispeicher** [comment]
I think the only way for reading dataframes with entire null columns is to explicitly specify all other columns and therefore ignore the null columns. For your  example this would look like this:
```python
import awswrangler as wr
df = wr.s3.read_parquet(path='s3://<S3_BUCKET>/df_null_columns.parquet', boto3_session=boto3.Session())
gen = wr.s3.read_parquet(path='s3://<S3_BUCKET>/df_null_columns.parquet', chunked=True,boto3_session=boto3.Session(), columns=["col"])
next(gen)
>>   col
  0    1
  1    2
```

However this means you would need to know beforehand which columns are null which you can do by running:
```python
wr.s3.read_parquet_metadata(path="s3://<S3_BUCKET>/df_null_columns.parquet")
>> ({'col': 'bigint', 'col_string_null': 'null', 'col_date_null': 'null'}, None)
```

**Comment by Natalie-Caruana** [comment]
Thanks alot for this!

**Comment by igorborgest** [comment]
I can see some space for improvements here. This schema inference that is failing to find out the null type is only necessary if the argument `validate_schema=True` (non default). So the commit above fix it and will be shipped on version `2.4.0`.
